components/supervisor/pkg/ports/served-ports.go

```diff
@@ -87,19 +87,35 @@ func (p *PollingServedPortsObserver) Observe(ctx context.Context) (<-chan []Serv
 					errchan <- err
 					continue
 				}
-				ports = append(ports, ps...)
+				for _, port := range ps {
+					current, exists := ports[port.Port]
+					if exists {
+						if current.BoundToLocalhost && !port.BoundToLocalhost {
+							current.BoundToLocalhost = false
+						}
+					} else {
+						if ports == nil {
```

Comment:
Why not move initialization closer to declaration? Maybe `ports := make(map[uint32]*ServedPort)`?